test/test_tags.c

```diff
@@ -726,7 +726,7 @@ void test_tag_at_limit(void) {
 
   assert(tag_name_len - 1 == NOIT_TAG_MAX_PAIR_LEN);
 
-  noit_metric_tags_parse_one(tag_name, tag_name_len - 1, &tag, &too_long);
+  noit_metric_tags_parse_one(tag_name, tag_name_len - 1, &tag, &too_long, NOIT_TAG_MAX_PAIR_LEN);
```

Comment:
Explicitly asked for a new function so that we wouldn't need to change the call sites... Furthermore, unless this tag is ``__name``, this is allowing tags that are way too big.  Same for the other call sites

Reply:
this function is not public and is called by `noit_metric_tagset_builder_add_one` and `noit_metric_tagset_builder_add_name`

Reply:
I can rename it to make it clear that it is not exposed

Reply:
No, that's fine, as long the bounds are properly checked

Reply:
@pjulien `noit_metric_tags_parse_one` is in `noit_metric.h`. Why are you no longer concerned about changing the call site? There are no public repositories on github that call this function (aside from reconnoiter forks), but the function is still exposed. In updating this, are we assuming that no one else is calling this function?

Reply:
because you said it's private, hence it's private

Reply:
I was wrong about that; this is a public function (it is declared in the header).
I will try to figure out a way to implement this without changing the call to `noit_metric_tags_parse_one`

Reply:
Make a new function please, call the new function, with extra arguments, from the old functions